### PR TITLE
feat: build filters

### PR DESCRIPTION
**Pull request by mingard**

### CODE REVIEW

File: dadi/frontend/components/DocumentFilters/DocumentFilters.jsx [hunk 1]
```diff
@@ -19,52 +19,75 @@ export default class DocumentFilters extends Component {
     } = this.props
 
     // Evaluate passed filter / store in state
+    let filters = (isValidJSON(filter) ? objectToArray(JSON.parse(filter), 'field') : []).map(this.deconstructFilters)
     this.state = {
```
Comment (eduardoboucas) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: We shouldn't be mutating `state` directly, only with `setState()`.

File: dadi/frontend/components/DocumentFilters/DocumentFilters.jsx [hunk 2]
```diff
@@ -19,52 +19,75 @@ export default class DocumentFilters extends Component {
...
         )}
       </div>
     )
   }
 
-  updateUrlFilters(filter) {
+  deconstructFilters(filter) {
+    // If there is no filter type and the field value is a string, add a type
+    if (typeof filter.value === 'string') return Object.assign(filter, {type: '$eq'})
+    else {
```
Comment (eduardoboucas) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: This looks a bit odd. We usually put the body of the `if` on a new line if there is an `else` block, like:

```js
  deconstructFilters(filter) {
    // If there is no filter type and the field value is a string, add a type
    if (typeof filter.value === 'string') {
      return Object.assign(filter, {type: '$eq'})
    } else {
      let type = Object.keys(filter.value)[0]

      return Object.assign(filter, {type: type, value: filter.value[type]})
    }
  }
```

File: dadi/frontend/components/DocumentFilters/DocumentFilters.jsx (hunk 1 above)
Comment (mingard) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: @eduardoboucas sure, but this isn't

Comment (mingard) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: This is in pre-render. Just like https://github.com/dadi/publish/blob/07adcb74e9766f3fb1bc85894dec675bf4565121/dadi/frontend/components/FieldImage/FieldImage.jsx#L13-L16

Comment (mingard) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: I have updated this, but suggest we do it everywhere it we really want to treat state as immutable

**Pull request merged**